src/systemctl/systemctl.c

```diff
@@ -7517,8 +7569,17 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
 
                         if (arg_full)
                                 r = unit_file_create_copy(&lp, unit_name, path, &new_path, &tmp_path);
-                        else
-                                r = unit_file_create_new(&lp, unit_name, ".d/override.conf", &new_path, &tmp_path);
+                        else {
+                                _cleanup_free_ char *path_copy = strdup(path);
+                                if (!path_copy)
+                                        return log_oom();
+
+                                r = strv_consume_prepend(&unit_paths, TAKE_PTR(path_copy));
```

Comment:
instead of copying locally, we should just have strv_prepend() that copies for us

Reply:
I've added a `strv_prepend` which duplicates `strv_extend` but calls `strv_consume_extend` instead of `srv_consume`.  Any particular place they should go in strv.c/h? I've put them both just before `strv_extend`.